### PR TITLE
DBTP-459 Add svc deploy command to deploy specific image

### PR DESCRIPTION
**Pull request by WillGibson**
https://uktrade.atlassian.net/browse/DBTP-459

Deploying a specific image was apiece of cake with `IMAGE_TAG=commit-abc1234 copilot svc deploy --env dev --name web`.

However there was a desire that developers should still be able to use latest, without having to think about which exact tag it was, then the command use the commit hash tag behind the scenes, which is what this does.

Example usages:

* `copilot-helper svc deploy --env dev --name=web --repository demodjango --image-tag c7654f23`
* `copilot-helper svc deploy --env dev --name=web --repository demodjango --image-tag 1.0.0`
* `copilot-helper svc deploy --env dev --name=web --repository demodjango --image-tag latest`
* `copilot-helper svc deploy --env dev --name=web --repository demodjango`

### CODE REVIEW

File: dbt_copilot_helper/COMMANDS.md [hunk 1]
```diff
@@ -589,6 +592,53 @@ copilot-helper domain assign-domain --app <app> --env <env> --svc <svc>
...
+
+## Options
+
+- `--env <text>`
+
+- `--name <text>`
+
+- `--repository <text>`
+
+- `--image-tag <text>` _Defaults to latest._
```
Comment (WillGibson) [inline dbt_copilot_helper/COMMANDS.md:637]: I went to --image-tag` over simply `--tag` as AWS Copilot uses the latter for other things and I wanted it to be distinct.

File: dbt_copilot_helper/commands/svc.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            )
+            return list(filtered)[0]
+        except IndexError:
+            click.secho(
+                """The image tagged "latest" does not have a commit tag.""",
+                fg="red",
+            )
+            exit(1)
+
+    image_tags = get_all_tags_for_image(image_tag)
```
Comment (antroy-madetech) [inline dbt_copilot_helper/commands/svc.py:61]: Looks like this can go in the if block below to save unnecessary calls to AWS

Comment (antroy-madetech) [reply dbt_copilot_helper/commands/svc.py:61]: Or is it for validation that the tag specified is an actual tag?

File: pyproject.toml [hunk 3]
```diff
@@ -45,6 +45,7 @@ pytest-env = "^0.8.1"
 tox = "^4.6.0"
 pre-commit = "^3.3.3"
 freezegun = "^1.2.2"
+faker = "^19.6.2"
```
Comment (yusufsheiqh) [inline pyproject.toml:48]: Is this being used anywhere?

File: dbt_copilot_helper/commands/svc.py (hunk 2 above)
Comment (WillGibson) [reply dbt_copilot_helper/commands/svc.py:61]: Yes, if the tag cannot be found it catches the exception and exists with the failure message.

**Review by antroy-madetech — APPROVED** [review]
Minor comment that I'll leave up to you to use or not. Otherwise all good.

File: dbt_copilot_helper/commands/svc.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                f"""No image exists with the tag "{image_tag_needle}".""",
+                fg="red",
+            )
+            exit(1)
+
+    def get_commit_tag_for_latest_image(image_tags_haystack):
+        try:
+            filtered = filter(
+                lambda tag: re.match("(commit-[a-f0-9]{7,32})", tag), image_tags_haystack
+            )
```
[the remark this answers is not in the document]
Comment (WillGibson) [reply dbt_copilot_helper/commands/svc.py]: Thanks @antroy-madetech. Done.

**Pull request merged**